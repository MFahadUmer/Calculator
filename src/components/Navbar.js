import React from 'react';
import { Link, Route, Switch } from 'react-router-dom';
import App from './App';
import Quote from './Quote';
import Home from './Home';

const Navbar = () => (
  <div className="navbarContainer">
    <Switch>
      <Route exact path="/" component={Home} />
      <Route path="/calculator" component={App} />
      <Route path="/quote" component={Quote} />
    </Switch>
    <div>
      <h2>Math Magicians</h2>
    </div>
    <div>
      <Link to="/">Home </Link>
      <Link to="/calculator">Calculator </Link>
      <Link to="/quote">Quote of the Day </Link>
    </div>
  </div>
);

export default Navbar;
