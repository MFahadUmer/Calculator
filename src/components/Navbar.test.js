import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Navbar from './Navbar';

let container = null;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('contains 3 links with different urls', () => {
  render(
    <BrowserRouter>
      <Navbar />
    </BrowserRouter>,
    container,
  );
  expect(screen.getByText('Home').href).toBe('http://localhost/');
  expect(screen.getByText('Calculator').href).toBe('http://localhost/calculator');
  expect(screen.getByText('Quote Of The Day').href).toBe('http://localhost/quote');
});

it('renders correctly', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
