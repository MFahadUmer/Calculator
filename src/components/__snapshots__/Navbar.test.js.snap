// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="navbarContainer"
>
  <div>
    <h2
      className="title"
    >
      Math Magicians
    </h2>
  </div>
  <div>
    <a
      className="links"
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
    <span
      className="verticalLine"
    >
      |
    </span>
    <a
      className="links"
      href="/calculator"
      onClick={[Function]}
    >
      Calculator
    </a>
    <span
      className="verticalLine"
    >
      |
    </span>
    <a
      className="links"
      href="/quote"
      onClick={[Function]}
    >
      Quote Of The Day
    </a>
  </div>
</div>
`;
